source/module_relax/relax_new/relax.cpp

```diff
@@ -411,7 +411,8 @@ void Relax::new_direction()
     //prepare for line search
     bool restart = true;
     double x=0, y=etot;
-    double xnew, yd;
+    //TODO: add a certain threshold for the progress.
+    double xnew, yd = 1e-8;
```

Comment:
is there a reason why yd=1e-8?

Reply:
`line_search` requires a parameter `conv_thr`. As seen in functions `third_order` and `brent`, `conv_thr` is used to judge convergence. Therefore, the value shall be >0 and small enough, and `1e-8` is picked for initialization.